opta/core/cloud_client.py

```diff
@@ -28,3 +28,7 @@ def delete_remote_state(self) -> None:
     @abstractmethod
     def get_terraform_lock_id(self) -> str:
         raise NotImplementedError()
+
+    @classmethod
+    def get_detailed_config_map(cls) -> Dict[str, Dict[str, str]]:
```

Comment:
since we already have
```
    @abstractmethod
    def get_remote_config(self) -> Optional["StructuredConfig"]:
        pass
```

let's call this new method using a similar name
```
    @abstractmethod
    def list_remote_configs(self) -> Dict[str, "StructuredConfig"]]:
       #TODO add docstring
        pass
```
list_ is a better prefix some similar name to `s3.list_objects`

We should be able to reuse the implementation done `get_remote_config` by creating a new private method   `def _get_remote_config(layer_name str) that could be called by both `get_remote_config` and `list_remote_configs` . That would avoid duplicating some similar code. 


it would be up to the view command call `["original_spec"]` when displaying the info to the user

Reply:
I'd rather we have these 2 separate functionality as of now...because they both are serving a different purpose. Also, adding `["original_spec"]` to the `view` isn't clean. If at any point we decide to change how we store config, we'll have to make changes to the command, which isn't required.
Also, `get_remote_config` serves 2 purposes,  one is checking the version of opta with which the commands were run, and another is getting details for the modules for the new terraform generation purpose.
As of now, what we can do is make the download part common for the both of them, as that's the only functionality which I see to be common among the both of them.

Reply:
Also, we can extend this functionality at a later stage when we just don't need the actual configs but the lists of configurations or some other functionality.

Reply:
> Also, get_remote_config serves 2 purposes, one is checking the version of opta with which the commands were run, and another is getting details for the modules for the new terraform generation purpose.

What is the part with checking the version, I don't see it in the impl https://github.com/run-x/opta/blob/main/opta/core/aws.py#L78-L90

I have an issue with this name `get_detailed_config_map` because it sounds like k8s ConfigMap. And in the existing code we already use the names `remote_configs` and `original_spec`.

What you can do also is:
```
    @abstractmethod
    def list_remote_configs(self) -> Dict[str, "StructuredConfig"]]:
       #TODO add docstring
        pass

   def list_original_spec(self) -> Dict[str, str]]:
       #TODO add docstring
        # implement using list_remote_configs  and return ["original_spec"]
```

The method `list_original_spec` would be implemented in one place.